### DISCUSSION TITLE
Reading Thai Documents

### DISCUSSION THREAD

**Discussion by ghost** [pymupdf/PyMuPDF#3264] [General]
Hi everyone,
I was wondering if there is a parameter I can tinker with to get better reading of Thai encoded documents? When I use the traditional methods I am getting  this weird symbol '�':
<img width="650" alt="image" src="https://github.com/pymupdf/PyMuPDF/assets/80626325/e952cd7f-1b9d-4731-a279-26f85f2afe93">

Do you have a suggestion? Thanks in advance!

**Comment by JorjMcKie** [comment]
Which PyMuPDF version are you using?
Some version ago, the base library added logic to automatically use the glyph number whenever the font returned the Invalid Unicode `�`.

**Reply by ghost** [reply]
Hello Jorj,
Thanks for your answer.
These are the versions we tried: 
PyMuPDF                    1.20.1 
PyMuPDF                     1.23.26

**Comment by ghost** [comment]
PyMuPDF 1.24.0 got worse:

<img width="806" alt="image" src="https://github.com/pymupdf/PyMuPDF/assets/80626325/a69215e0-d204-4808-a168-f78281f1115c">

**Reply by JorjMcKie** [reply]
of course I need an example file to follow up ...
The current version is 1.23.26 btw

**Reply by ghost** [reply]
Hi,
Thanks again Jorj, sending it:
[SCB_442726092.pdf](https://github.com/pymupdf/PyMuPDF/files/14634267/SCB_442726092.pdf)

**Reply by JorjMcKie** [reply]
Tried it out: none of my PDF-to-text converters can successfully handle this file.
Sorry.

**Reply by ghost** [reply]
Thanks for your quick answer! Will see about other ways.

Best
